puz/Puzzle.cpp

```diff
@@ -568,4 +569,60 @@ const Puzzle::FileHandlerDesc Puzzle::sm_saveHandlers[] = {
     { NULL, NULL, NULL }
 };
 
+// -----------------------------------------------------------------------
+// Diagramless
+// -----------------------------------------------------------------------
+
+void
+Puzzle::FixMalformattedDiagramless()
+{
+    if (!IsDiagramless()) return;
+
+    // If the timer was started on this puzzle, do nothing.
+    if (GetTime() != 0) return;
+
+    // Check if the grid has perfect black squares; that is, the "user" has entered precisely the
+    // correct set of black squares, but the grid is otherwise blank.
+    bool grid_has_perfect_blacks = true;
+    for (Square * square = GetGrid().First();
+         square != NULL;
+         square = square->Next())
+    {
+        if ((square->IsSolutionBlack() != square->IsBlack())
+            || (!square->IsBlack() && !square->IsBlank())) {

```

Comment:
I'm continuing to regret the choice to use 'blank' since it looks so much like 'black'